DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -57,31 +96,47 @@ function Get-TargetResource
             $($LocalizedData.FirewallRuleDoesNotExistMessage) -f $Name
             ) -join '')
         return @{
-            Name   = $Name
             Ensure = 'Absent'
+            Name   = $Name
         }
     }
 
     $properties = Get-FirewallRuleProperty -FirewallRule $firewallRule
 
-    # Populate the properties for get target resource
-    return @{
-        Name            = $Name
-        Ensure          = 'Present'
-        DisplayName     = $firewallRule.DisplayName
-        Group           = $firewallRule.Group
-        DisplayGroup    = $firewallRule.DisplayGroup
-        Enabled         = $firewallRule.Enabled
-        Action          = $firewallRule.Action
-        Profile         = $firewallRule.Profile.ToString() -replace(' ', '') -split(',')
-        Direction       = $firewallRule.Direction
-        Description     = $firewallRule.Description
-        RemotePort      = @($properties.PortFilters.RemotePort)
-        LocalPort       = @($properties.PortFilters.LocalPort)
-        Protocol        = $properties.PortFilters.Protocol
-        Program         = $properties.ApplicationFilters.Program
-        Service         = $properties.ServiceFilters.Service
+    $Result = @{
+        Ensure = 'Present'
+    }
+
+    # Populate the properties for get target resource by looping through
+    # the parameter array list and adding the values to 
+    foreach ($p in $ParameterList)
```

Comment:
Could we call this foreach ($parameter in $parameterlist) so it's clearer what we're iterating on?